app/rest/metrics-prometheus/src/main/java/io/syndesis/rest/metrics/prometheus/PrometheusMetricsProviderImpl.java

```diff
@@ -15,60 +15,102 @@
-            "io_syndesis_camel_LastExchangeCompletedTimestamp", "syndesis_io_deployment_id", Date.class, maxDate);
-
-        Optional<Long> totalMessages = totalMessagesMap.map(Map::values).flatMap(
-            longs -> Optional.of(longs.stream().mapToLong(Long::longValue).sum()));
-        Optional<Long> failedMessages = failedMessagesMap.map(Map::values).flatMap(
-            longs -> Optional.of(longs.stream().mapToLong(Long::longValue).sum()));
-        Optional<Date> startTime = startTimeMap.map(Map::values).flatMap(
-            dates -> dates.stream().max(Date::compareTo));
-        Optional<Date> lastProcessingTime = lastProcessingTimeMap.map(Map::values).flatMap(
-            dates -> dates.stream().max(Date::compareTo));
+
+        final Map<String, Long> totalMessagesMap = getMetricValues(integrationId,
+            "org_apache_camel_ExchangesTotal", deploymentVersionLabel, Long.class, SUM_LONGS);
+        final Map<String, Long> failedMessagesMap = getMetricValues(integrationId,
+            "org_apache_camel_ExchangesFailed", deploymentVersionLabel, Long.class, SUM_LONGS);
+
+        final Map<String, Date> startTimeMap = getMetricValues(integrationId,
+            "io_syndesis_camel_StartTimestamp", deploymentVersionLabel, Date.class, MAX_DATE);
+        final Map<String, Date> lastProcessingTimeMap = getMetricValues(integrationId,
+            "io_syndesis_camel_LastExchangeCompletedTimestamp", deploymentVersionLabel, Date.class, MAX_DATE);
+
+        return createIntegrationMetricsSummary(totalMessagesMap, failedMessagesMap, startTimeMap,
+            lastProcessingTimeMap);
+    }
+
+    private IntegrationMetricsSummary createIntegrationMetricsSummary(Map<String, Long> totalMessagesMap,
+                                                                      Map<String, Long> failedMessagesMap,
+                                                                      Map<String, Date> startTimeMap,
+                                                                      Map<String, Date> lastProcessingTimeMap) {
+
+        final long[] totalMessages = {0L};
+        final long[] totalErrors = {0L};
+        final Date[] startTime = { null };
+        final Date[] lastProcessedTime = { null };
+
+        final List<IntegrationDeploymentMetrics> deploymentMetrics = totalMessagesMap.entrySet().stream().map(entry -> {
+            final String version = entry.getKey();
+            final Long messages = entry.getValue();
+
+            final Long errors = failedMessagesMap.get(version);
+            final Date start = startTimeMap.get(version);
+            final Date lastProcessed = lastProcessingTimeMap.get(version);
+
+            // aggregate values while we are at it
+            totalMessages[0] = SUM_LONGS.apply(totalMessages[0], messages);
+            totalErrors[0] = SUM_LONGS.apply(totalErrors[0], errors);
+            startTime[0] = MAX_DATE.apply(startTime[0], start);
```

Comment:
Does that really show the date of how long the integration has been up?

Reply:
@KurtStam it shows the latest start date, isn't that what we agreed upon?

Reply:
@dhirajsb was just wondering if we need to nullify that when the pod goes down (upon a new deployment version)

Reply:
@KurtStam I think it's still valid to show when a particular version had last started. If we want to show integration pod status, like Prometheus we should use another metric e.g. `up` that shows whether a monitored resource is active or not. 
That way, both the last start time and up/down status is captured for every deployment version. Otherwise there is no way to report start times for older versions, right?

Reply:
@dhirajsb - I switched to 'starttime' so we can more easily cache it; uptime is something that changes every second.. This way we only don't need to write metrics if no messages flow through the system. Uptime implies it's running though imh.

Reply:
@KurtStam if the ui knows whether the current deployment is running, it can compute the 'uptime' for the integration from the start date. Is that not what it's doing? 
It's not a good idea to use Date to hold a duration anyway, there is a `java.time.Duration` type for that and it's obvious what it's for.